9651 scalar current date (#10978)
crate/crate: docs/general/ddl/data-types.rst

    +-----------------------+
    SELECT 1 row in set (... sec)

.. _date-data-type:

``date``
---------

.. NOTE::

    The storage of the ``date`` data type is not supported. Therefore,
    it is not possible to create tables with ``date`` fields.

> The ``date`` data type consist of year, month, and day.
> A constant value of type ``date`` can be expressed by explicit casts 
> of string or numeric literals.
> 
> The string format for dates is `YYYY-MM-DD`. For example `2021-03-09`.

Reviewer: ```suggestion
The ``date`` data type can be used to represent values with a year, month and a day.

To construct values of the type ``date`` you can cast a string literal or a numeric literal to ``date``.
If a numeric value is used, it must contain the number of milliseconds since ``1970-01-01T00:00:00Z``.

The string format for dates is `YYYY-MM-DD`. For example `2021-03-09`.
```

    +-----------------------+
    SELECT 1 row in set (... sec)

.. _date-data-type:

> date
> -------------------

Reviewer 2: ```suggestion
``date``
---------
```

> A string literal that contain a date will be
> converted to UTC date.

Reviewer 2: ```suggestion
.. NOTE::

    The storage of the ``date`` data type is not supported. Therefore,
    it is not possible to create tables with ``date`` fields.

The ``date`` data type consist of year, month, and day.
A constant value of type ``date`` can be expressed by explicit casts 
of string or numeric literals.

The string format for dates is `YYYY-MM-DD`. For example `2021-03-09`.
```

::

    cr> select '2021-03-09'::date as cd;
    +---------------+
    |            cd |
    +---------------+
    | 1615248000000 |
    +---------------+
    SELECT 1 row in set (... sec)

Date also accepts a ``bigint`` representing UTC milliseconds since
the epoch.

::

    cr> select 1615248000000::date as cd;
   +---------------+
    |            cd |
    +---------------+
    | 1615248000000 |
    +---------------+

>     SELECT 1 row in set (... sec)

Reviewer 2: I'd remove this example if you take my suggestion above which already mention the support of numeric->date casts.

Author: Done. I removed string casting example `select '2021-03-09'::date` as well as it's also mentioned in suggestion.

Reviewer 2: I think keeping the string cast example would still be valuable (also this may be the common use case).

Author: done

    +-------------------------------+
    | result                        |
    +-------------------------------+

>     +-------------------------------+

Reviewer 2: Mistake? ;)

Author: yes, conflicts resolving mistake, thanks for spotting it.
